### PR TITLE
chore: upgrade Rust to nightly 2022-12-20

### PR DESCRIPTION
**Pull request by MichaelScofield**
I hereby agree to the terms of the [GreptimeDB CLA](https://gist.github.com/xtang/6378857777706e568c1949c7578592cc)

## What's changed and what's your intention?

I want to use workspace dependency feature introduced in newer version of cargo, otherwise upgrading DataFusion is painful. Upgrade to last night's Rust, still can't use stable. Modified some codes as clippy suggested accordingly.

## Checklist

- [ ]  I have written the necessary rustdoc comments.
- [ ]  I have added the necessary unit tests and integration tests.

### CODE REVIEW

**Review by waynexia — APPROVED** [review]
LGTM

File: src/meta-srv/src/service/heartbeat.rs [hunk 1]
```diff
@@ -100,7 +100,7 @@ impl heartbeat_server::Heartbeat for MetaSrv {
                 pusher_key.as_ref().unwrap_or(&"unknow".to_string())
             );
             if let Some(key) = pusher_key {
-                let _ = handler_group.unregister(&key);
+                let _ = handler_group.unregister(&key).await;
```
Comment (evenyag) [inline src/meta-srv/src/service/heartbeat.rs:103]: What did clippy say? Looks like `.await` is needed.

File: src/query/src/optimizer.rs [hunk 2]
```diff
@@ -34,19 +34,15 @@ use datatypes::arrow::datatypes::DataType;
 pub struct TypeConversionRule;
 
 impl OptimizerRule for TypeConversionRule {
-    fn optimize(
-        &self,
-        plan: &LogicalPlan,
-        optimizer_config: &mut OptimizerConfig,
-    ) -> Result<LogicalPlan> {
+    fn optimize(&self, plan: &LogicalPlan, _config: &mut OptimizerConfig) -> Result<LogicalPlan> {
```
Comment (evenyag) [inline src/query/src/optimizer.rs:37]: Since `_config` is used in the following codes, we should rename it to `config`.

Comment (MichaelScofield) [reply src/query/src/optimizer.rs:37]: clippy complained with https://rust-lang.github.io/rust-clippy/master/index.html#only_used_in_recursion

File: src/meta-srv/src/service/heartbeat.rs (hunk 1 above)
Comment (MichaelScofield) [reply src/meta-srv/src/service/heartbeat.rs:103]: https://rust-lang.github.io/rust-clippy/master/index.html#let_underscore_future

File: src/query/src/optimizer.rs (hunk 2 above)
Comment (evenyag) [reply src/query/src/optimizer.rs:37]: Thanks, I get it. Looks like we could move the optimization logic to a function which only takes the plan as input, so we don't need to passing the config to it. Let's defer it and merge this PR first.

**Review by evenyag — APPROVED** [review]
LGTM

**Pull request merged**